### PR TITLE
Initial linux version

### PR DESCRIPTION
**Pull request by aetherical**
Here's an initial linux version. I added an extra flag beyond the OS/X version -- the ability to specify the device.  Another difference is the checking to see if wget and/or s3 binaries exist....

I don't know that this will work with every flavour of linux, but I tried to make it as generic as possible.

### CODE REVIEW

File: flash.linux [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+image=$1
+
+occi=$2
+
+if [[ "$1" == "--help" ]]; then
+    usage
+fi
+
+if beginswith http:// "$image" ;then
+    which curl 2>/dev/null || error "Error: curl not found. Aborting" 1
```
Comment (StefanScherer) [inline flash.linux]: Nice detail. I'll add this kind of check into the Mac version as well!

**Comment by StefanScherer** [comment]
Thanks for your contribution!
I'll merge soon.

**Comment by Govinda-Fichtner** [comment]
What about keeping the options/interface for the flash scripts the same? If we have an option to specify the device in linux we should have that for mac, too. What do you think guys?

**Comment by aetherical** [comment]
I think it's a good idea to (as much as possible) to keep them in sync.  

Another thing I'm going to add on my end is an (optional) flag to do an md5 of what was written to verify. I've run into a couple bad cards in the last day or so :-/. Had I'd been able able to test it might have saved time.....

**Pull request merged**